src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/vmware/VMWareProviderVirtualMachineUtil.java

```diff
@@ -84,9 +72,110 @@ public VirtualMachine searchVirtualMachineByName(String name, Folder rootFolder)
 
     }
 
-    public Folder searchFolderByName(String name, Folder rootFolder)
-            throws InvalidProperty, RuntimeFault, RemoteException {
+    public Folder searchFolderByName(String name, Folder rootFolder) throws RemoteException {
         return (Folder) new InventoryNavigator(rootFolder).searchManagedEntity("Folder", name);
     }
 
+    public HostSystem searchHostByName(String name, Folder rootFolder) throws RemoteException {
+        return (HostSystem) new InventoryNavigator(rootFolder).searchManagedEntity("HostSystem", name);
+    }
+
+    public Folder searchVMFolderFromVMName(String name, Folder rootFolder) throws RemoteException {
+        ManagedEntity current = searchVirtualMachineByName(name, rootFolder).getParent();
+        while (!(current instanceof Folder)) {
+            current = current.getParent();
+        }
+        return (Folder) current;
+    }
+
+    public Folder searchVMFolderByHostname(String name, Folder rootFolder) throws RemoteException {
+        return Lists.newArrayList(new InventoryNavigator(rootFolder).searchManagedEntities("Folder"))
+                    .stream()
```

Comment:
I created an enum, but seems weird the way I use it, don't know if its the best way to do it
see 29a118819496c5464cc0fd26ad78da39f73a0fa2